Dockerfile

```diff
@@ -34,6 +39,23 @@ RUN set -ex; \
     chmod +x /usr/local/bin/frpc; \
     rm -rf /tmp/frp /tmp/frp.tar.gz
 
+#############################
+# Stage 2: Final Runtime Image
+#############################
+FROM python:3.12-slim-bookworm
+
+# Install any runtime apt packages your app needs.
+RUN apt-get update && apt-get install -y curl procps && \
+    apt-get clean && rm -rf /var/lib/apt/lists/*
+
+# Copy installed Python packages and FRP client from the builder.
+COPY --from=builder /usr/local/ /usr/local/
+# Copy application files and scripts from the builder.
+COPY --from=builder /ex_app/ /ex_app/
+COPY --from=builder /healthcheck.sh /healthcheck.sh
+COPY --from=builder /start.sh /start.sh
```

Comment:
wdyt of adding the external files only once at this place? It could reduce one copy step.
FRP install too since it places the binary files and does not have any leftovers.